src/pages/Earn/UnstakeModal.tsx

```diff
@@ -0,0 +1,173 @@
+import { Trans, t } from "@lingui/macro";
+import { ethers } from "ethers";
+import React, { useState } from "react";
+import RewardRouter from "sdk/abis/RewardRouter.json";
+import { ARBITRUM } from "config/chains";
+import { BASIS_POINTS_DIVISOR_BIGINT } from "config/factors";
+import { getIcons } from "config/icons";
+import { SetPendingTransactions } from "domain/legacy";
+import { bigMath } from "lib/bigmath";
+import { callContract } from "lib/contracts";
+import { ProcessedData } from "lib/legacy";
+import { formatAmount, formatAmountFree, parseValue } from "lib/numbers";
+import { UncheckedJsonRpcSigner } from "lib/rpc/UncheckedJsonRpcSigner";
+import Button from "components/Button/Button";
+import Modal from "components/Modal/Modal";
+import { AlertInfo } from "components/AlertInfo/AlertInfo";
+import BuyInputSection from "components/BuyInputSection/BuyInputSection";
+
+export function UnstakeModal(props: {
+  isVisible: boolean;
+  setIsVisible: (isVisible: boolean) => void;
+  chainId: number;
+  title: string;
+  maxAmount: bigint | undefined;
+  value: string;
+  setValue: (value: string) => void;
+  signer: UncheckedJsonRpcSigner | undefined;
+  unstakingTokenSymbol: string;
+  rewardRouterAddress: string;
+  unstakeMethodName: string;
+  reservedAmount: bigint | undefined;
+  setPendingTxns: SetPendingTransactions;
+  processedData: ProcessedData | undefined;
+}) {
+  const {
+    isVisible,
+    setIsVisible,
+    chainId,
+    title,
+    maxAmount,
+    value,
+    setValue,
+    signer,
+    unstakingTokenSymbol,
+    rewardRouterAddress,
+    unstakeMethodName,
+    reservedAmount,
+    setPendingTxns,
+    processedData,
+  } = props;
+  const [isUnstaking, setIsUnstaking] = useState(false);
+  const icons = getIcons(chainId);
+
+  let amount = parseValue(value, 18);
+
+  let unstakeBonusLostPercentage: undefined | bigint = undefined;
+  if (
+    processedData &&
+    amount !== undefined &&
+    amount > 0 &&
+    processedData.esGmxInStakedGmx !== undefined &&
+    processedData.gmxInStakedGmx !== undefined
+  ) {
+    const divisor = processedData.esGmxInStakedGmx + processedData.gmxInStakedGmx;
+    if (divisor !== 0n) {
+      unstakeBonusLostPercentage = bigMath.mulDiv(amount, BASIS_POINTS_DIVISOR_BIGINT, divisor);
+    }
+  }
+
+  const votingPowerBurnAmount = amount;
+
+  const getError = () => {
```

Comment:
In this file also present some values which are calculated on every render which is redundant, it's better use `useMemo`

Reply:
done